src/status.rs

```diff
@@ -214,9 +214,23 @@ impl Status {
         header_map.insert(GRPC_STATUS_HEADER_CODE, self.code.to_header_value());
 
         if !self.message.is_empty() {
-            header_map.insert(GRPC_STATUS_MESSAGE_HEADER, HeaderValue::from_str(&self.message)
-                .map_err(invalid_header_value_byte)?);
+            let good_bytes = &self.message.as_bytes().iter().all(|&x| DEFAULT_ENCODE_SET.contains(x));
+            let to_write = 
+            match good_bytes{
+                true => {
+                    Bytes::from(
+                        percent_encode(&self.message().as_bytes(), DEFAULT_ENCODE_SET)
+                        .to_string()
+                        .as_bytes())
```

Comment:
This is a slight gotcha: when percent-encoding, we have a new `String` (from `to_string()`), which owns some memory. It's nearly free to convert a `String` into a `Bytes`, as it just assumes ownership of that memory. By calling `Bytes::from(String.as_bytes())`, this is asking `Bytes` to create a new slot of memory and copy, since `as_bytes` returns a borrow of the `String`s memory.

So, this could just be `percent_encode(..).to_string().into()` (type inference will detect what `into` will convert to because the other branch is also returning `Bytes`.

Reply:
Thanks! Learn a new way to use `into()`